codeguru_profiler_agent/sampling_utils.py

```diff
@@ -69,14 +70,16 @@ def _extract_stack(stack, max_depth):
         )
     if len(stack) < max_depth:
         last_frame, last_frame_line_no = stack[-1]
-        _maybe_append_time_sleep(result, last_frame, last_frame_line_no)
+        _maybe_append_synthetic_frame(result, last_frame, last_frame_line_no)
     return result
 
 
-def _maybe_append_time_sleep(result, frame, line_no):
+def _maybe_append_synthetic_frame(result, frame, line_no):
     line = linecache.getline(frame.f_code.co_filename, line_no).strip()
     if "sleep(" in line:
         result.append(TIME_SLEEP_FRAME)
+    elif "etree.XMLSchema" in line:
```

Comment:
1. should it be etree.XMLSchema( as we include the open bracket in sleep(
2. we would miss this if user had:

from etree import XMLSchema

a = XMLSchema()

Should we cover that case too and simply use XMLSchema? How likely would be gives us false positive in matching line like ComplexXMLSchema (<-- if that exists)?

Reply:
Good point about the parenthesis, will add it.

Yes you are right about missing out on `from etree import XMLSchema` but searching for XMLSchema could give many false positive and according to what I have seen in existing code I believe most people would do `etree.XMLSchema` as provided in the online documentation. I prefer to miss out on a few cases rather have false positives which could be very confusing.